cloudera/datadog_checks/cloudera/api_client_factory.py

```diff
@@ -0,0 +1,34 @@
+import cm_client
+import cm_client.rest
+import packaging.version
+
+from datadog_checks.base import ConfigurationError
+from datadog_checks.cloudera.api_client_v7 import ApiClientV7
+
+
+def make_api_client(check, config):
+    cm_client.configuration.username = config.workload_username
+    cm_client.configuration.password = config.workload_password
```

Comment:
This would be the global state of `cm_client`'s configuration. There's no current Github repo for the `cm_client` code but essentially the client looks at the `cm_client.configuration` to generate the header parameters when it is about to send a request. 

In `api_client.py`'s code:

```
    def update_params_for_auth(self, headers, querys, auth_settings):
        """
        Updates header and query params based on authentication setting.

        :param headers: Header parameters dict to be updated.
        :param querys: Query parameters tuple list to be updated.
        :param auth_settings: Authentication setting identifiers list.
        """
        config = Configuration().  # This returns a Configuration object containing the username and password

        if not auth_settings:
            return

        for auth in auth_settings:
            auth_setting = config.auth_settings().get(auth)
            if auth_setting:
                if not auth_setting['value']:
                    continue
                elif auth_setting['in'] == 'header':
                    headers[auth_setting['key']] = auth_setting['value']
                elif auth_setting['in'] == 'query':
                    querys.append((auth_setting['key'], auth_setting['value']))
                else:
                    raise ValueError(
                        'Authentication token must be in `query` or `header`'
                    )
```

Reply:
Right I see:

![cm_client](https://user-images.githubusercontent.com/9677399/208791867-98b91bd0-ba73-4eab-88bb-745baaaf0995.PNG)

Should we put the user/pass in `init_config` then?

Reply:
Ah yeah that's a good catch! I will move the credential configs to `init_config`.